src/AuthenticatedApiClient/index.js

```diff
@@ -0,0 +1,223 @@
+ * @property {function} options
+ * @property {function} delete (csrf protected)
+ * @property {function} post (csrf protected)
+ * @property {function} put (csrf protected)
+ * @property {function} patch (csrf protected)
+ */
+
+/**
+ * Gets the apiClient singleton which is an axios instance.
+ * 
+ * @param {object} config 
+ * @param {string} [config.appBaseUrl]
+ * @param {string} [config.authBaseUrl]
+ * @param {string} [config.loginUrl]
+ * @param {string} [config.logoutUrl]
+ * @param {function} [config.handleEmptyAccessToken] (optional)
+ * @param {object} [config.loggingService] requires logError and logInfo methods
+ * @param {string} [config.refreshAccessTokenEndpoint]
+ * @param {string} [config.accessTokenCookieName]
+ * @param {string} [config.csrfTokenApiPath]
+ * @returns {HttpClient} Singleton. A configured axios http client
+ */
+function getAuthenticatedApiClient(authConfig) {
+  if (authenticatedAPIClient === null) {
+    configure(authConfig);
+    authenticatedAPIClient = axios.create();
+
+    // Axios interceptors
+    const refreshAccessTokenInterceptor = jwtTokenProviderInterceptor({
+      tokenCookieName: config.accessTokenCookieName,
+      tokenRefreshEndpoint: config.refreshAccessTokenEndpoint,
+      handleEmptyToken: config.handleEmptyAccessToken,
+      handleUnexpectedRefreshError: handleUnexpectedAccessTokenRefreshError,
+      isExempt: axiosRequestConfig => axiosRequestConfig.isPublic,
+    });
+    const attachCsrfTokenInterceptor = csrfTokenProviderInterceptor({
+      csrfTokenApiPath: config.csrfTokenApiPath,
+      isExempt: (axiosRequestConfig) => {
+        const { method, isCsrfExempt } = axiosRequestConfig;
+        const CSRF_PROTECTED_METHODS = ['post', 'put', 'patch', 'delete'];
+        return isCsrfExempt || !CSRF_PROTECTED_METHODS.includes(method);
+      },
+    });
+
+    // Request interceptors: Axios runs the interceptors in reverse
+    // order from how they are listed. Since fetching csrf token does
+    // not require jwt authentication, it doesn't matter which
+    // happens first.
+    authenticatedAPIClient.interceptors.request.use(attachCsrfTokenInterceptor);
+    authenticatedAPIClient.interceptors.request.use(refreshAccessTokenInterceptor);
+
+    // Response interceptor: moves axios response error data into
+    // the error object at error.customAttributes
+    authenticatedAPIClient.interceptors.response.use(
+      response => response,
+      processAxiosRequestErrorInterceptor,
+    );
+  }
+
+  return authenticatedAPIClient;
+}
+
+/**
+ * @typedef UserAccessToken
+ * @property {string} userId
+ * @property {string} username
+ * @property {array} roles
+ * @property {bool} administrator
+ */
+
+/**
+ * Gets the authenticated user's access token. Resolves to null if the user is unauthenticated.
+ *
+ * @returns {Promise<UserAccessToken>|Promise<null>} Resolves to the user's access token if they are logged in.
+ */
+const getAuthenticatedUserAccessToken = async () => {
```

Comment:
Any reason we can't call this `getAuthenticatedUser`?  I don't know that it's important for consumers of this to think of it as an access token...

Reply:
Sounds good to me. Changed